feat: Adding Generic Forwarded Emails

jjack opened the pull request:
## Proposed change

Adding the ability to use a list of generic forwarded email addresses when searching for new emails.

## Type of change


- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation update
- [ ] Adds a new shipper
- [ ] Update existing shipper

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This change combines the existing Amazon list with the new Generic list when searching for new emails, it does not remove the existing Amazon list.
- This PR fixes or closes issue: fixes #1075

jjack committed:
chore: updating launch.json to use newer format and debugpy

jjack committed:
chore(deps): adding tox to dev install of home assistant

StephenBrown2 committed:
fix: adjust SSL contexts to use the non-cached versions provided by HomeAssistant (#1095)

firstof9 committed:
fix: filter out more ads for mail images (#1104)

jjack committed:
feat: adding ability to use forwarded email addresses

Refs: #1075

jjack committed:
chore: cleaning up missed formatting issues

firstof9 commented on custom_components/mail_and_packages/helpers.py:
```diff
@@ -1679,3 +1710,40 @@ def get_items(
     return value
+
+
+def generate_service_email_domains() -> set[str]:
```
`AMAZON_DOMAINS` is no longer used, the user enters their amazon email domains in during configuration. That way we're not looping excessively.

firstof9 commented on custom_components/mail_and_packages/helpers.py:
```diff
@@ -33,10 +33,12 @@
 from .const import (
     AMAZON_DELIVERED,
     AMAZON_DELIVERED_SUBJECT,
+    AMAZON_DOMAINS,
```
`AMAZON_DOMAINS` is no longer used, the user enters their amazon email domains in during configuration. That way we're not looping excessively.

firstof9 committed:
fix: add additional amazon delivery subject (#1110)

* fix: add additional amazon delivery subject

* update tests

firstof9 committed:
fix: add additional amazon delivery subject (#1110)

* fix: add additional amazon delivery subject

* update tests

jjack commented:
I'm terribly sorry for the delay on this. I've been out of town and the
last message got caught up in the fray.

I should be able to get to this sometime in the next few days.

On Wed, Jun 25, 2025, 10:07 PM github-actions[bot] ***@***.***>
wrote:

> *github-actions[bot]* left a comment
> (moralmunky/Home-Assistant-Mail-And-Packages#1109)
> <https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/pull/1109#issuecomment-3006883731>
>
> This PR has been automatically marked as stale because it has not had
> recent activity. It will be closed if no further activity occurs. Thank you
> for your contributions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/pull/1109#issuecomment-3006883731>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQQDXCEHSLYBCY6LZQOVLAL3FNPWHAVCNFSM6AAAAAB7DP5CH6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAMBWHA4DGNZTGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

jjack committed:
fix: using amazon domains provided by the user instead of AMAZON_DOMAINS

jjack commented:
I'm unfortunately at a bit of a loss on these failing tests. They're passing when run through VS Code and "pytest" but _not_ when run through tox's pytest. The environments seem to be a little bit different, with different versions of plugins. I'm not familiar with tox and I'm having some trouble figuring out why they're different or why they're failing at all.

The pytest under the vscode environment is 8.3.5 and all the tests pass.
    .venvvscode ➜ /workspaces/Home-Assistant-Mail-And-Packages (generic-forwarded-emails) $ pytest --fixtures
    Test session starts (platform: linux, Python 3.13.3, pytest 8.3.5, pytest-sugar 1.0.0)
    rootdir: /workspaces/Home-Assistant-Mail-And-Packages
    configfile: tox.ini
    plugins: homeassistant-custom-component-0.13.236, timeout-2.3.1, respx-0.22.0, unordered-0.6.1, picked-0.5.1, pytest_freezer-0.4.9, anyio-4.9.0, syrupy-4.8.1, github-actions-annotate-failures-0.3.0, cov-6.0.0, xdist-3.6.1, requests-mock-1.12.1, socket-0.7.0, aiohttp-1.1.0, asyncio-0.26.0, sugar-1.0.0
    asyncio: mode=Mode.AUTO, asyncio_default_fixture_loop_scope=function, asyncio_default_test_loop_scope=function
    collected 125 items

The pytest that tox runs is 8.4.0 and is having the issues.
    .venvvscode ➜ /workspaces/Home-Assistant-Mail-And-Packages (generic-forwarded-emails) $ /workspaces/Home-Assistant-Mail-And-Packages/.tox/py313/bin/pytest --fixtures
    Test session starts (platform: linux, Python 3.13.3, pytest 8.4.0, pytest-sugar 1.0.0)
    rootdir: /workspaces/Home-Assistant-Mail-And-Packages
    configfile: tox.ini
    plugins: respx-0.22.0, picked-0.5.1, timeout-2.4.0, syrupy-4.9.1, cov-6.1.1, pytest_freezer-0.4.9, anyio-4.9.0, asyncio-1.0.0, github-actions-annotate-failures-0.3.0, homeassistant-custom-component-0.13.257, xdist-3.7.0, unordered-0.7.0, requests-mock-1.12.1, socket-0.7.0, aiohttp-1.1.0, sugar-1.0.0
    asyncio: mode=Mode.AUTO, asyncio_default_fixture_loop_scope=function, asyncio_default_test_loop_scope=function
    collected 125 items

firstof9 commented:
Try adding `turbojpeg` to `requirements_test.txt` seems to be what's tripping some of it up.

ChrisLanks committed:
fix: Update Amazon sensors for delivery and shipped (#1114)

* Update Amazon sensors for delivery and shipped

The regex needed to include single words for "today" to be detected, and the deliver email has changed to the following format:
```
Message-ID: <<long-uuid>@email.amazonses.com>
Subject: Delivered: 2 items | Order # 111-9196457-791234
MIME-Version: 1.0
```

* Update requirements_test.txt

---------

Co-authored-by: Chris <1105672+firstof9@users.noreply.github.com>

ChrisLanks committed:
fix: Update Amazon sensors for delivery and shipped (#1114)

* Update Amazon sensors for delivery and shipped

The regex needed to include single words for "today" to be detected, and the deliver email has changed to the following format:
```
Message-ID: <<long-uuid>@email.amazonses.com>
Subject: Delivered: 2 items | Order # 111-9196457-791234
MIME-Version: 1.0
```

* Update requirements_test.txt

---------

Co-authored-by: Chris <1105672+firstof9@users.noreply.github.com>

firstof9 commented:
Can you run `black custom_components/` for the formatting and push the commit?

firstof9 committed:
Merge branch 'dev' into generic-forwarded-emails

jjack committed:
fix: swapping turbojpeg for PyTurboJPEG

jjack committed:
fix: swapping turbojpeg for PyTurboJPEG

jjack commented:
I got a chance to poke around at this in detail and it tuns out that Home Assistant uses a different wrapper for libturbojpeg. I've updated the requirements to use the one from HASS.

firstof9 commented:
Looks like you'll need to rebase.

jjack committed:
fix: rebasing

jjack committed:
Merge branch 'generic-forwarded-emails' of github.com:jjack/Home-Assistant-Mail-And-Packages into generic-forwarded-emails

jjack commented:
Closing this PR to make a better, cleaner branch.
